src/layout/layout.rs

```diff
@@ -217,6 +220,7 @@ impl Layout {
     /// * is called before [`Layout::split()`] otherwise, the cache size is
     ///   [`Self::DEFAULT_CACHE_SIZE`].
     /// * is called for the first time, subsequent calls do not modify the cache size.
+    #[allow(clippy::must_use_candidate)] // TODO: should return Result?
     pub fn init_cache(cache_size: usize) -> bool {
```

Comment:
This return value isn't always used… And somehow I think it should be handled as it's basically a `Result::is_ok -> bool`.

Personally I think this is a bug that should be fixed in its own PR?